### PR TITLE
add security2 module with saml2 (+introduce lombok)

### PR DESCRIPTION
**Pull request by domivds**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -413,6 +418,22 @@ project('security') {
   }
 }
 
+project('security2') {
```
Comment (natan-cox-cognizone) [inline build.gradle:421]: naming by an engineer 😏

File: security2/src/main/java/zone/cogni/asquare/security2/saml2/Saml2Properties.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+@Data
+public class Saml2Properties {
+
+  private SigningKeys signingKeyStore;
+  private String registrationId;
+  private String idpUrl;
+  private Attributes attributes;
+  private String roleMappingUrl;
+  private Map<String,User> basicAuthUsers = new HashMap<>(); //init so we allow empty config
```
Comment (natan-cox-cognizone) [inline security2/src/main/java/zone/cogni/asquare/security2/saml2/Saml2Properties.java]:
```suggestion
  private Map<String, User> basicAuthUsers = new HashMap<>(); //init so we allow empty config
```

File: security2/src/main/java/zone/cogni/asquare/security2/saml2/Saml2HttpConfigurer.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+public class Saml2HttpConfigurer extends AbstractHttpConfigurer<Saml2HttpConfigurer, HttpSecurity> {
```
Comment (natan-cox-cognizone) [inline security2/src/main/java/zone/cogni/asquare/security2/saml2/Saml2HttpConfigurer.java:33]: sure this is only picked up if we enable saml?

**Review by natan-cox-cognizone — APPROVED** [review]

File: security2/src/main/java/zone/cogni/asquare/security2/saml2/Saml2HttpConfigurer.java (hunk 3 above)
Comment (domivds) [reply security2/src/main/java/zone/cogni/asquare/security2/saml2/Saml2HttpConfigurer.java:33]: as long as this is not "bean-ized" somewhere we are ok

**Pull request merged**